packages/ai/src/ai/web/middleware.py

```
@@ -1,39 +1,95 @@
+        """
+        request = Request(scope, receive, send)
 
+        try:
+            error_response = await server.authenticate_request(request, return_tuple=True)
+            if error_response:
+                # Reject WebSocket — send HTTP 403 before upgrade
+                response = Response(status_code=403, content='Authentication failed')
+                await response(scope, receive, send)
+                return
+            await self.app(scope, receive, send)
         except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)